cloudify_agent/operations.py

```diff
@@ -195,17 +195,14 @@ def _get_broker_url(ctx, agent):
 
 
 def create_new_agent_dict(old_agent):
-    new_agent = copy.deepcopy(old_agent)
-    fields_to_delete = ['name', 'queue', 'workdir', 'agent_dir', 'envdir',
-                        'manager_ip', 'package_url', 'source_url'
-                        'manager_port', 'broker_url', 'broker_port',
-                        'broker_ip', 'key', 'version']
-    for field in fields_to_delete:
-        if field in new_agent:
-            del(new_agent[field])
+    new_agent = {}
     new_agent['name'] = utils.internal.generate_new_agent_name(
         old_agent['name'])
     new_agent['remote_execution'] = True
+    fields_to_copy = ['windows', 'ip', 'basedir', 'user']
```

Comment:
when do we actually need to preserve any old property? if we simply call `reinstallation_attributes` with an empty initial new_agent dict, should that suffice?
The runtime property gets populated in the first place from similar methods in the configuration module.